File: hw_4_objects/java/ObjectsTest.java

// Note on the TemplateTest:
//
// You really only need to do two things:
//
// 1. Search/replace TemplateTest with the name of your test.
//
// 2. Implement a bunch of functions that start with test and end with
// the name of the test you want. The names should match those
// provided in ../description-common.json. For example if a test is
// FooThingWorks, your test function should be testFooThingWorks.
//
// Nothing else is required as long as you attach the RetroPrinter to
// your test runner.

import org.junit.Test;
import org.junit.Before;
import static org.junit.Assert.*;
import org.junit.runner.JUnitCore;

public class ObjectsTest {

    public static void main(String[] args) {
	JUnitCore core= new JUnitCore();
	core.addListener(new RetroPrinter());
	core.run(ObjectsTest.class);
    }

    Student[] many;
    Objects objects;

    private Student initStudent(String n, int a, String m) {
	Student s = new Student();
	s.name = n;
	s.age = a;
	s.major = m;
	return s;
    }

    @Before
    public void init() {
	objects = new Objects();
	many = new Student[] {
	    initStudent("James Bond", 38, "Bartending"),
	    initStudent("Charlie Parker", 50, "Music"),
	    initStudent("Marky Mark", 20, "Music"),
	    initStudent("Bob Bobson", 21, "Landscape Architecture")
	};
    }

    @Test 
    public void testMakeStudent() {
        String n = "Geronimo";
	int a = 40;
        String m = "Landscape Architecture";
        Student s = objects.makeStudent(n, a, m);
        assertEquals(n, s.name);
        assertEquals(a, s.age);
        assertEquals(m, s.major);
    }

    @Test 
    public void testMakeStudentOlder() {
	for (Student s : many) {
	    int a = s.age;
	    objects.makeStudentOlder(s);
	    assertEquals(a+1, s.age);
	}
    }
    
    @Test 
    public void testCopyStudent() {
	for (Student s : many) {
	    Student s_copy = objects.copyStudent(s);
	    assertNotNull(s_copy);
            assertTrue(s_copy.name == s.name);
            assertTrue(s_copy.age == s.age);
            assertTrue(s_copy.major == s.major);
            assertFalse(s_copy == s);
	}
    }
    
    @Test 
    public void testIsOlder() {
        assertTrue(objects.isOlder(many[0]));  // 38
	assertTrue(objects.isOlder(many[1]));  // 50
        assertFalse(objects.isOlder(many[2])); // 20
        assertFalse(objects.isOlder(many[3])); // 21
    }

    @Test 
    public void testIsMusician() {
        assertFalse(objects.isMusician(many[0]));
        assertTrue(objects.isMusician(many[1]));
        assertTrue(objects.isMusician(many[2]));
        assertFalse(objects.isMusician(many[3]));
    }

    @Test 
    public void testIsOlderMusician() {
        assertFalse(objects.isOlderMusician(many[0]));
        assertTrue(objects.isOlderMusician(many[1]));
        assertFalse(objects.isOlderMusician(many[2]));
        assertFalse(objects.isOlderMusician(many[3]));
    }


}

File: hw_4_objects/py/objects_test.py
import unittest
from RetroPrinter import RetroPrinter
from sys import argv

from objects import *

class RetroTest(unittest.TestCase):

    def __init__(self, methodName, outputName):
        super(RetroTest, self).__init__(methodName) # call superconstructor
        self.outputName = outputName

    def setUp(self):
        pass

    def tearDown(self):
        pass

    def init_student(self, n, a, m):
        s = Student()
        s.name = n
        s.age = a
        s.major = m
        return s

    def init_students(self):
        ret = []
        ret.append(self.init_student("James Bond", 38, "Bartending"))
        ret.append(self.init_student("Charlie Parker", 50, "Music"))
        ret.append(self.init_student("Marky Mark", 20, "Music"))
        ret.append(self.init_student("Bob Bobson", 21, "Landscape Architecture"))
        return ret

    def test_make_student(self):
        n = "Geronimo"
        a = 40
        m = "Landscape Architecture"
        s = make_student(n, a, m)
        self.assertEqual(n, s.name)
        self.assertEqual(a, s.age)
        self.assertEqual(m, s.major)

    def test_make_student_older(self):
        many = self.init_students()
        for s in many:
            a = s.age
            make_student_older(s)
            self.assertEqual(a+1, s.age, "Student did not get older")

    def test_copy_student(self):
        many = self.init_students()
        for s in many:
            s_copy = copy_student(s)
            self.assertIsNotNone(s_copy)
            self.assertTrue(s_copy.name == s.name)
            self.assertTrue(s_copy.age == s.age)
            self.assertTrue(s_copy.major == s.major)
            self.assertFalse(s_copy is s)

    def test_is_older(self):
        many = self.init_students()
        self.assertTrue(is_older(many[0])) # 38
        self.assertTrue(is_older(many[1])) # 50
        self.assertFalse(is_older(many[2])) # 20
        self.assertFalse(is_older(many[3])) # 21

    def test_is_musician(self):
        many = self.init_students()
        self.assertFalse(is_musician(many[0]))
        self.assertTrue(is_musician(many[1]))
        self.assertTrue(is_musician(many[2]))
        self.assertFalse(is_musician(many[3]))

    def test_is_older_musician(self):
        many = self.init_students()
        self.assertFalse(is_older_musician(many[0]))
        self.assertTrue(is_older_musician(many[1]))
        self.assertFalse(is_older_musician(many[2]))
        self.assertFalse(is_older_musician(many[3]))


if __name__ == '__main__':
    retro = False
    if (len(argv) > 1 and argv[1] == "--retrograde"):
        retro = True
    print "Retro mode: " + str(retro)
    suite = unittest.TestSuite()
    # now pair the above functions to the retrograde name
    tests = [
        RetroTest("test_make_student", "MakeStudent"),
        RetroTest("test_make_student_older", "MakeStudentOlder"),
        RetroTest("test_copy_student", "CopyStudent"),
        RetroTest("test_is_older", "IsOlder"),
        RetroTest("test_is_musician", "IsMusician"),
        RetroTest("test_is_older_musician", "IsOlderMusician"),
        ]
    for test in tests:
        suite.addTest(test)
    if retro:
        result = RetroPrinter()
        print "Running Python test suite..."
        suite.run(result)
        print "Done running Python test suite."
    else:
        runner = unittest.TextTestRunner()
        runner.run(suite)
